src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2971,6 +3046,45 @@ def update_sku(self, mc: ManagedCluster) -> ManagedCluster:
             )
         return mc
 
+    def update_upgrade_settings(self, mc: ManagedCluster) -> ManagedCluster:
+        """Update upgrade settings for the ManagedCluster object.
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        # There is a limitation on differentiating empty list vs. not set in update requests.
+        # In such case, we'll use a workaround here to disable it by setting the until field to the current time, to make the overrides no longer effective.
+        # For now there's only one allowed override so we can return early here.
+        if self.context.get_disable_upgrade_ignore_kubernetes_deprecations() == True:
+            if mc.upgrade_settings.override_settings.control_plane_overrides == [CONST_IGNORE_KUBERNETES_DEPRECATIONS]:
```

Comment:
- better to check nils for the sub-profiles
- and not sure if it's better to check `.control_plane_overrides` contains (instead of equal to) `CONST_IGNORE_KUBERNETES_DEPRECATIONS`. The order also affects the result of equal to, although it seems currently only one value is supported in this list..

Reply:
We only set until to now() if the only value in mc.upgrade_settings.override_settings.control_plane_overrides is CONST_IGNORE_KUBERNETES_DEPRECATIONS.
Updated.